models/users.models.js

```diff
@@ -1,12 +1,45 @@
 const db = require('../db/connection.js');
 const format = require('pg-format');
+const errors = require('../errors.js');
 
 exports.selectUsers = () => {
 
     const query = `SELECT username FROM users;`;
 
     return db.query(query)
         .then(res => {
-            return res.rows;
+            return res.rows; 
         });
-}
\ No newline at end of file
+}
+
+exports.selectUserByUsername = async (username) => {
+
+    return await this._selectByProperty('username', username);
+};
+
+
+
+
+/*******************************************************
+ * PRIVATE METHODS
+ *******************************************************/
+
+exports._selectByProperty = async (property, value) => {
```

Comment:
This is really neat. I've not seen such flexible handling of queries before. Good job!